### PR TITLE
rgw/pubsub: send notifications from multi-delete op

### PR DESCRIPTION
**Pull request by yuvalif**
Signed-off-by: Yuval Lifshitz <yuvalif@yahoo.com>

Fix: https://tracker.ceph.com/issues/43214

> Note that to support metadata in multi-delete request (as well as delete request) this: https://github.com/ceph/ceph/pull/31878 has to be merged first

### CODE REVIEW

**Comment by yuvalif** [comment]
jenkins test make check arm64

**Comment by yuvalif** [comment]
teuthology: http://pulpito.front.sepia.ceph.com/yuvalif-2020-01-19_16:00:06-rgw:verify-wip-yuval-multidelete-notifications-distro-basic-smithi/

failures are similar issues to: https://github.com/ceph/ceph/pull/32618#issuecomment-575989136

multisite teuthology: http://pulpito.front.sepia.ceph.com/yuvalif-2020-01-19_16:01:31-rgw:multisite-wip-yuval-multidelete-notifications-distro-basic-smithi/
(only non-pubsub multisite are failing)

File: src/rgw/rgw_op.cc [hunk 1]
```diff
@@ -6514,6 +6514,22 @@ void RGWDeleteMultiObj::execute()
 
     send_partial_response(*iter, del_op.result.delete_marker,
 			  del_op.result.version_id, op_ret);
+
+    const auto obj_state = obj_ctx->get_state(obj);
+    s->object = obj.key;
```
Comment (mattbenjamin) [inline src/rgw/rgw_op.cc]: aren't we in a range-for over 1..m objects here?  I haven't read older versions of this code that I remember, but it seems I'm surprised we would assigned req_state's object or size members here.  I don't see where in the current PR this is really doing anything (other than letting some state leak out of this routine)?

Comment (yuvalif) [reply src/rgw/rgw_op.cc]: agree it is a bit of a hack, but the `publish` call already expect req_state as a parameter.
other option would be:
- add key and size as explicit parameters to `publish`
- create a local copy of req_state and set it with key and size

Comment (yuvalif) [reply src/rgw/rgw_op.cc]: @mattbenjamin done, added name and size as parameters to `publish`

**Pull request merged**